pkg/daemon/daemon.go

```diff
@@ -497,11 +497,26 @@ func (dn *Daemon) nodeStateSyncHandler(generation int64) error {
 		}
 	}
 
-	if reqDrain && !dn.disableDrain {
-		glog.Info("nodeStateSyncHandler(): drain node")
-		if err := dn.drainNode(dn.name); err != nil {
+	if reqDrain {
+		ctx, cancel := context.WithCancel(context.TODO())
+		defer cancel()
+
+		glog.Infof("nodeStateSyncHandler(): get drain lock for sriov daemon")
+		done := make(chan bool)
+		go dn.getDrainLock(ctx, done)
+		<-done
+
+		glog.Infof("nodeStateSyncHandler(): pause MCP")
+		if err := dn.pauseMCP(); err != nil {
 			return err
 		}
+
+		if !dn.disableDrain {
```

Comment:
It is better to keep this check consistent between function `drainNode` and `completeDrain`. Either put this check in both the functions or out of both.

Reply:
done can you please have another look?